velocity/tool/src/templates/VM_chef_library.vm

```diff
@@ -645,3 +645,109 @@ scheduleUpdate();
+	return true;
+}
+function doPageNav(url, buttonElement)
+{
+	SPNR.disableControlsAndSpin(buttonElement, null);
+	location = url;
+	return true;
+}
+</script>
+<div class="sakai-table-pagerLabel">$tlang.getFormattedMessage("gen.viewing.phrase", $topMsgPos, $btmMsgPos, $allMsgNumber)</div>
+<div class="sakai-table-pagerControls">
+	<span class="skip">$tlang.getString("gen.first")</span>
+	<input type="button" name="eventSubmit_doList_first" value="&#124;&lt;" title="$tlang.getString("gen.first")"
+		onclick="doPageNav('#toolLink("$action", "doList_first")', this);" #if ($goFPButton != "true") disabled="disabled" #end />
+
+	<span class="skip">$tlang.getFormattedMessage('gen.previous.pagesize', $pagesize)</span>
+	<input type="button" name="eventSubmit_doList_prev" value="&lt;" title="$tlang.getFormattedMessage('gen.previous.pagesize', $pagesize)"
+		onclick="doPageNav('#toolLink("$action", "doList_prev")', this);" #if ($goPPButton != "true") disabled="disabled" #end />
+
+	<label for="selectPageSize_$panelName">$tlang.getString("gen.select.label")</label>
+	<span class="skip">$tlang.getString("gen.listnavselect")</span>
+	<div class="sakai-table-pagerPageSize">
+		<select id="selectPageSize_$panelName" name="selectPageSize_$panelName" onchange="doChangePageSize('#toolLink("$action", "doChange_pagesize")', this);">
+		#foreach ($i in $!pagesizes)
+			<option value="$i" #if($pagesize == $i) selected="selected" #end>$tlang.getFormattedMessage("gen.show", $i)</option>
+		#end
+		</select>
+	</div>
+
+	<span class="skip">$tlang.getFormattedMessage('gen.next.pagesize', $pagesize)</span>
+	<input type="button" name="eventSubmit_doList_next" value="&gt;" title="$tlang.getFormattedMessage('gen.next.pagesize', $pagesize)"
+		onclick="doPageNav('#toolLink("$action", "doList_next")', this);" #if($goNPButton != "true") disabled="disabled" #end />
+
+	<span class="skip">$tlang.getString('gen.last')</span>
+	<input type="button" name="eventSubmit_doList_last" value="&gt;&#124;" title="$tlang.getString('gen.last')"
+		onclick="doPageNav('#toolLink("$action", "doList_last")', this);" #if ($goLPButton != "true") disabled="disabled" #end />
+</div>
+#end ## pagerPanel
+
+#macro (searchFilterPanel $panelName $userSearchTerm $searchActionKey $clearSearchActionKey)
+<div class="sakai-table-searchFilter">
+	<label for="$panelName">$tlang.getString("search.label")</label>
+	<div class="sakai-table-searchFilterControls">
+		<input type="text" name="$panelName" id="$panelName" value="$validator.escapeHtml($!userSearchTerm)" class="searchField" />
+		<input type="button" id="btnSearch_$panelName" value="$tlang.getString('search.button')" onclick="doSearch('#toolLink("$action" "$searchActionKey")', 'btnSearch_$panelName', '$panelName');" />
+		#if($!userSearchTerm && !$userSearchTerm.isEmpty())
+			<input type="button" value="$tlang.getString("search.clear")" onclick="doClearSearch('#toolLink("$action" "$clearSearchActionKey")', this);" />
+		#end
+	</div>
```

Comment:
This is in a velocity macro that is intended to be on many different pages. Inlining the <script> required by the macro seems reasonable to me given I'm not aware of a good mechanism in velocity to let macros contribute to the page header.

Reply:
doesn't every page in Sakai load the morpheus scripts js ?

Reply:
That would put all these Velocity-specific scripts on every page in Sakai when they aren't needed.

Reply:
hundreds of small scripts are optimized, stripped, packed and loaded on every page .... cached by browser so hard to argue that putting scripts into a central place is bad for page load performance.

Reply:
Until we reach the breaking point where hundreds of unnecessary scripts being loaded on every page becomes a problem. I'd rather not contribute to inefficiency for the sake of appeasing codacy.

Reply:
Please, don't put inline JavaScript code, I agree with @ottenhoff on this should be on morpheus, ping me if you need anything

Reply:
You can add a class on this input and call it from $PBJQ('.class').on('click',...);

Reply:
I think the only thing codacity is flagging is because the script is inline in the file. A few other places in this file include external js

	## Everything is in an our own external JavaScript library
	<script type="text/javascript" src="#libraryLink('calendar/sakai-calendar.js')"></script>